homeassistant/components/habitica/config_flow.py

```diff
@@ -374,3 +388,68 @@ async def validate_api_key(
             return errors, user.data
 
         return errors, None
+
+    @classmethod
+    @callback
+    def async_get_supported_subentry_types(
+        cls, config_entry: ConfigEntry
+    ) -> dict[str, type[ConfigSubentryFlow]]:
+        """Return subentries supported by this integration."""
+        return {"party_member": PartyMembersSubentryFlowHandler}
+
+
+class PartyMembersSubentryFlowHandler(ConfigSubentryFlow):
+    """Handle subentry flow for adding party members."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> SubentryFlowResult:
+        """Subentry user flow."""
+
+        entry: HabiticaConfigEntry = self._get_entry()
+        if entry.state is not ConfigEntryState.LOADED:
+            return self.async_abort(reason="config_entry_disabled")
+        if (party := entry.runtime_data.data.user.party.id) is None:
+            return self.async_abort(reason="not_in_a_party")
+
+        party_members = self.hass.data[HABITICA_KEY][party].data.members
+
+        if user_input is not None:
+            config_entries = self.hass.config_entries.async_entries(DOMAIN)
+            if user_input[CONF_PARTY_MEMBER] in {
+                entry.unique_id for entry in config_entries
+            }:
+                return self.async_abort(reason="already_configured_as_entry")
```

Comment:
lets move this check down to the for-loop on 424, so that we don't have to iterate twice on the config_entries.